Tests/LagrangeTest/LagrangeUnitTest.cpp

```diff
@@ -0,0 +1,183 @@
+        std::vector<double> interp_x(num_points);
+        std::vector<double> interp_y(num_points);
+        std::vector<double> interp_z(num_points);
+
+        double extract_radius = sim_params.L / 4;
+
+        for (int ipoint = 0; ipoint < num_points; ++ipoint)
+        {
+            double phi   = ipoint * 2. * M_PI / num_points;
+            double theta = ipoint * M_PI / num_points;
+            interp_x[ipoint] =
+                sim_params.center[0] + extract_radius * cos(phi) * sin(theta);
+            interp_y[ipoint] =
+                sim_params.center[1] + extract_radius * sin(phi) * sin(theta);
+            interp_z[ipoint] =
+                sim_params.center[2] + extract_radius * cos(theta);
+        }
+
+        // std::cout << "Interp_x[0] " << interp_x[0] - sim_params.center[0] <<
+        // std::endl; std::cout << "Interp_y[0] " << interp_y[0] -
+        // sim_params.center[1] << std::endl; std::cout << "Interp_z[0] " <<
+        // interp_z[0] - sim_params.center[2] << std::endl;
+
+        // set-up query
+        InterpolationQueryParticle query(num_points);
+        query.setCoords(0, interp_x.data())
+            .setCoords(1, interp_y.data())
+            .setCoords(2, interp_z.data())
+            .addComp(0, A.data(), Derivative::LOCAL, VariableType::derived);
+
+        // set up interpolation using Particles
+        ParticleInterpolators<1> interpolator(sim_params.boundary_params, 0);
+        interpolator.set_gramr_ptr(&gr_amr);
+        interpolator.populate_from_query(query);
+
+        std::vector<const amrex::MultiFab *> fields;
+        fields.reserve(poly_by_lev.size());
+        for (const auto &mf_uptr : poly_by_lev)
+        {
+            fields.push_back(
+                mf_uptr.get()); // convert to const amrex::MultiFab* to feed to
+                                // interp call like below
+        }
+
+        std::vector<int> comps(fields.size(),
+                               0); // we have only one comp, which is 0
+
+        interpolator.interpolate_to_particle_from_derived_fields(fields);
+
+        // Do not forget to set the parity!
+        interpolator.set_derived_var_parity(0, BCParity::even);
+
+        interpolator.interp(query, VariableType::derived);
+
+        if (amrex::ParallelDescriptor::MyProc() == 0)
+        {
+            double diff = 0; // absolute error
+
+            for (int ipoint = 0; ipoint < num_points; ++ipoint)
+            {
+                double x = interp_x[ipoint] - sim_params.center[0];
+                double y = interp_y[ipoint] - sim_params.center[1];
+                double z = interp_z[ipoint] - sim_params.center[2];
+
+                double value_A = 42. + x * x + y * y * z * z;
+
+                diff = fabs(A[ipoint] - value_A);
+
+                amrex::Print() << "Absolute error is " << std::setprecision(10)
+                               << diff << "\n";
+
+                CHECK(diff == doctest::Approx(0.0).epsilon(1e-10));
```

Comment:
Comparing the actual values rather than `diff` is more informative.
```suggestion
                CHECK(A[ipoint] == doctest::Approx(A_known).epsilon(1e-10));
```